webportal/webapp/src/services/cognito.service.ts

```diff
@@ -133,7 +133,18 @@ export class CognitoService {
                 idToken = session.getIdToken().getJwtToken()
             });
         } else {
-            console.log("NO TOKEN")
+            console.log("NO TOKEN - Retrieving new session.");
+            // If no token is available, clear localStorage and attempt to get a new session
+            // Generally, this appears to result in a logout, which seems OK
+            localStorage.clear();
+            var userAuth = new CognitoAuth(CognitoService._AUTH_DATA);
+            userAuth.userhandler = {
+                onSuccess: function(result) {
+                },
+                onFailure: function(err) {
+                }
+            };
+            userAuth.getSession();
```

Comment:
This seems to effectively do a safe log out. Seems like an OK thing to do if you have no token? Otherwise things don't work properly.

@aaronFranssell Any concerns with this?

Reply:
Nope no concerns... Our logout process is borked anyways...